ethereum/rpc/namespaces/miner/api.go

```diff
@@ -0,0 +1,195 @@
+	ethAPI  *eth.PublicAPI
+	backend backend.Backend
+}
+
+// NewMinerAPI creates an instance of the Miner API.
+func NewMinerAPI(
+	ctx *server.Context,
+	ethAPI *eth.PublicAPI,
+	backend backend.Backend,
+) *API {
+	return &API{
+		ctx:     ctx,
+		ethAPI:  ethAPI,
+		logger:  ctx.Logger.With("module", "miner"),
+		backend: backend,
+	}
+}
+
+// SetEtherbase sets the etherbase of the miner
+func (api *API) SetEtherbase(etherbase common.Address) bool {
+	api.logger.Debug("miner_setEtherbase")
+
+	delAddr, err := api.backend.GetCoinbase()
+	if err != nil {
+		api.logger.Debug("failed to get coinbase address", "error", err.Error())
+		return false
+	}
+
+	withdrawAddr := sdk.AccAddress(etherbase.Bytes())
+	msg := distributiontypes.NewMsgSetWithdrawAddress(delAddr, withdrawAddr)
+
+	if err := msg.ValidateBasic(); err != nil {
+		api.logger.Debug("tx failed basic validation", "error", err.Error())
+		return false
+	}
+
+	// Assemble transaction from fields
+	builder, ok := api.ethAPI.ClientCtx().TxConfig.NewTxBuilder().(authtx.ExtensionOptionsTxBuilder)
+	if !ok {
+		api.logger.Debug("clientCtx.TxConfig.NewTxBuilder returns unsupported builder", "error", err.Error())
+	}
+
+	err = builder.SetMsgs(msg)
+	if err != nil {
+		api.logger.Error("builder.SetMsgs failed", "error", err.Error())
+	}
+
+	denom, err := sdk.GetBaseDenom()
+	if err != nil {
+		api.logger.Debug("Could not get the denom of smallest unit registered.")
+		return false
+	}
+
+	delCommonAddr := common.BytesToAddress(delAddr.Bytes())
+	nonce, err := api.ethAPI.GetTransactionCount(delCommonAddr, rpctypes.EthPendingBlockNumber)
+	if err != nil {
+		api.logger.Debug("failed to get nonce", "error", err.Error())
+		return false
+	}
+
+	txFactory := tx.Factory{}
+	txFactory = txFactory.
+		WithChainID(api.ethAPI.ClientCtx().ChainID).
+		WithKeybase(api.ethAPI.ClientCtx().Keyring).
+		WithTxConfig(api.ethAPI.ClientCtx().TxConfig).
+		WithSequence(uint64(*nonce)).
+		WithGasAdjustment(1.11)
+
+	_, gas, err := tx.CalculateGas(api.ethAPI.ClientCtx(), txFactory, msg)
+	if err != nil {
+		api.logger.Debug("failed to calculate gas", "error", err.Error())
+		return false
+	}
+
+	txFactory = txFactory.WithGas(gas)
+
+	// Fetch minimun gas price to calculate fees using the configuration.
+	appConf, err := config.ParseConfig(api.ctx.Viper)
+	if err != nil {
+		api.logger.Error("failed to parse file.", "file", api.ctx.Viper.ConfigFileUsed(), "error:", err.Error())
+		return false
+	}
+
+	minGasPrices := appConf.GetMinGasPrices()
+	if len(minGasPrices) == 0 || minGasPrices.Empty() {
+		api.logger.Debug("the minimun fee is not set")
+		return false
+	}
+	minGasPriceValue := minGasPrices[0].Amount
+
+	value := new(big.Int).SetUint64(gas * minGasPriceValue.Ceil().TruncateInt().Uint64())
+	fees := sdk.Coins{sdk.NewCoin(denom, sdk.NewIntFromBigInt(value))}
+	builder.SetFeeAmount(fees)
+	builder.SetGasLimit(gas)
+
+	keyInfo, err := api.ethAPI.ClientCtx().Keyring.KeyByAddress(delAddr)
+	if err != nil {
```

Comment:
log error (debug)

Reply:
Added the log "failed to get the wallet address using the keyring"